### PR TITLE
Reworked GUI

### PR DESCRIPTION
**Pull request by SorataYuu**
Implemented an entirely new GUI. Also implemented sorting of Events and Tasks.

TO-DO: Work on the Today Pane, which is currently a WIP.

### CODE REVIEW

File: src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import javafx.beans.property.ObjectProperty;
 import javafx.beans.property.SimpleObjectProperty;
 
-public class DeadlineTask extends Task implements Comparable<DeadlineTask> {
+public class DeadlineTask extends Task implements Comparable<Task> {
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java]: Hmm why compare to Task? Task may not have dateTime. Shouldn't it be compare to DeadlineTask only?

File: src/main/java/cs2103/v15_1j/jimjim/ui/JJUI.java [hunk 2]
```diff
@@ -18,8 +15,12 @@
 	private final String APPLICATION_NAME = "JimJim";
 
 	public JJUI() {
-		mainViewController = new MainViewController();
-		mainViewController.setUIController(this);
+
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/JJUI.java]: empty function? Can it be removed?

File: src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java (hunk 1 above)
Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java]: I think it has something to do with inheritance. The Compiler was complaining that it vouldn't implement Comparable DeadlineTask cause the parent implemented Comparable Task

Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java]: I think we should research more about this before accepting this temporary solution.

Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java]: OKay. I see where the problem is. I think the best solution is to remove Comparable from Task and move it to Deadline and FloatingTask.

Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/model/DeadlineTask.java]: Shifted to Floating Task

**Pull request merged**